astyle: new version 4.1.11 and build using cmake

cessenat opened the pull request:
This is a major modification of the "astyle" package which shifted from makefile to cmake at 3.2.1
Compatibility with makefile versions is ensured.

Some interrogations I have, related to the package or more general to spack, I don't know:
- Despite it is a cmake package, and depends_on("cmake") is set, it may not work without setting ^cmake.
- As well "spack versions" does not benefit from url_for_version.

I would be pleased to have a review from @alalazo

cessenat committed:
astyle: new version 4.1.11 and build using cmake

cessenat committed:
astyle: correct style

cessenat commented:
I do not understand the style error reported by black.

tldahlgren commented on var/spack/repos/builtin/packages/astyle/package.py:
```diff
@@ -3,34 +3,65 @@
+
+    def url_for_version(self, version):
+        root = self.url.rsplit("/", 2)[0]
+        if version >= Version('3.2.0'):
+            url = "{0}/astyle%20{1}/astyle-{2}.tar.bz2"
+        else:
+            url = "{0}/astyle%20{1}/astyle_{2}_linux.tar.gz"
+        return url.format(root, version.up_to(2), version)
```
Optional.  We're slowly converting the core to use Python f-strings (see https://docs.python.org/3/tutorial/inputoutput.html) in case you want to see if that works here.

cessenat replied:
Thanks for the pointer.
I could do a
```
    def url_for_version(self, version):
        root = self.url.rsplit("/", 2)[0]
        if version >= Version("3.2.0"):
            return f"{root}/astyle%20{version.up_to(2)}/astyle-{version}.tar.bz2"
        else:
            return f"{root}/astyle%20{version.up_to(2)}/astyle_{version}_linux.tar.gz"

```
but I do not know how to set only one return, which I find more readable.

alalazo replied:
Maybe I am misunderstanding the issue, but wouldn't:
```python
    def url_for_version(self, version):
        root = self.url.rsplit("/", 2)[0]
        url = f"{root}/astyle%20{version.up_to(2)}/astyle-{version}.tar.bz2"
        if version < Version("3.2.0"):
            url = f"{root}/astyle%20{version.up_to(2)}/astyle_{version}_linux.tar.gz"
        return url
```
do it?

tldahlgren reviewed (dismissed):
Confirmed the new version sha256s.

alalazo commented:
@spackbot fix style

tldahlgren commented:
> I do not understand the style error reported by black.

Black wants double quotes for strings.

cessenat committed:
[@spackbot] updating style on behalf of cessenat

cessenat committed:
Update package.py from @alalazo advice

Using Python f-strings as suggested by @tldahlgren and then using the update written by @alalazo.

alalazo commented on var/spack/repos/builtin/packages/astyle/package.py:
```diff
@@ -3,34 +3,62 @@
 #
 # SPDX-License-Identifier: (Apache-2.0 OR MIT)
 
+from spack.build_systems.cmake import CMakeBuilder
+from spack.build_systems.makefile import MakefileBuilder
```
```suggestion
from spack.build_systems import makefile
```

alalazo commented on var/spack/repos/builtin/packages/astyle/package.py:
```diff
@@ -3,34 +3,62 @@
+        return url
+
+
+class CMakeBuilder(CMakeBuilder):
+    def cmake_args(self):
+        args = []
+        return args
+
+
```
I think this isn't necessary until you need to override something:
```suggestion
```

alalazo commented on var/spack/repos/builtin/packages/astyle/package.py:
```diff
@@ -3,34 +3,62 @@
+        return args
+
+
+class MakefileBuilder(MakefileBuilder):
```
```suggestion
class MakefileBuilder(makefile.MakefileBuilder):
```

alalazo reviewed (commented):
I left a few suggestions for cleanup. Beware that I didn't try the code, so typos might be in there :slightly_smiling_face:

cessenat committed:
Update var/spack/repos/builtin/packages/astyle/package.py

Co-authored-by: Massimiliano Culpo <massimiliano.culpo@gmail.com>

cessenat committed:
Update var/spack/repos/builtin/packages/astyle/package.py

Co-authored-by: Massimiliano Culpo <massimiliano.culpo@gmail.com>

cessenat committed:
Update var/spack/repos/builtin/packages/astyle/package.py

Co-authored-by: Massimiliano Culpo <massimiliano.culpo@gmail.com>
